manifests/bond.pp

```diff
@@ -1,163 +1,92 @@
+#   LACPDU packets in 802.3ad mode. Only applicable when mode is '802.3ad'.
+#   Options: slow/0 (every 30 seconds), fast/1 (every 1 second)
+# @param mtu
+#   The Maximum Transmission Unit size to use for bond interface and all slaves
+# @param options
+#   Hash with custom interfaces options
+# @param slave_options
+#   Hash with custom slave interfaces options
+# @param mode
+#   The interface bond mode. The value can be either the human readable term,
+#   such as 'active-backup' or the numeric representation, such as '1'.
+#   Options: balance-rr/0, active-backup/1, balance-xor/2, broadcast/3,
+#   802.3ad/4, balance-tlb/5, balance-alb/6
+# @param miimon
+#   The MII link monitoring frequency in milliseconds
+# @param downdelay
+#   The time in milliseconds, to wait before disabling a slave after a link
+#   failure has been detected
+# @param updelay
+#   The time in milliseconds to wait before enabling a slave after a link
+#   recovery has been detected
+# @param primary
+#   The primary interface to use when the mode is 'active-backup'. All other
+#   interfaces will be offline by default. Only applicable when mode is '802.3ad'
+# @param primary_reselect
+#   Specifies the reselection policy for the primary slave. This affects how
+#   the primary slave is chosen to become the active slave when failure of the
+#   active slave or recovery of the primary slave occurs. This option is
+#   designed to prevent flip-flopping between the primary slave and other slaves.
+#   Options: always/0, better/1, failure/2
+# @param xmit_hash_policy
+#   Selects the transmit hash policy to use for slave selection in balance-xor
+#   and 802.3ad modes. Options: layer2, layer2+3, layer3+4
+#
+# @example
+#   network::bond { 'bond0':
+#     ipaddress => '172.16.1.2',
+#     netmask   => '255.255.128.0',
+#     ensure    => present,
+#     slaves    => ['eth0', 'eth1'],
+#   }
+#
+# @see https://www.kernel.org/doc/Documentation/networking/bonding.txt Linux Ethernet Bonding Driver HOWTO, Section 2 "Bonding Driver Options"
 #
 define network::bond (
-  $slaves,
-  $ensure           = present,
-  $ipaddress        = undef,
-  $netmask          = undef,
-  $method           = undef,
-  $family           = undef,
-  $onboot           = undef,
-  $hotplug          = undef,
-  $lacp_rate        = undef,
-  $mtu              = undef,
-  $options          = undef,
-  $slave_options    = undef,
-
-  $mode             = 'active-backup',
-  $miimon           = '100',
-  $downdelay        = '200',
-  $updelay          = '200',
-  $primary          = $slaves[0],
-  $primary_reselect = 'always',
-  $xmit_hash_policy = 'layer2',
+  Array[String[1]]                                   $slaves,
+  Stdlib::Ensure::Package                            $ensure             = present,
+  Optional[Stdlib::IP::Address::V4]                  $ipaddress          = undef,
+  Optional[String[1]]                                $netmask            = undef,
```

Comment:
This an be Stdlib::IP::Address::Nosubnet also, I guess.. though not 100% sure. For ipv4 the mask is just an IP address at least.

Reply:
Here I meant that `$netmask` can be validated against `Stdlib::IP::Address::Nosubnet` (at least for IPv4, need to recheck for IPv6, but I believe it's the same). Though, I can accept String[1] for now also.